### PR TITLE
Release/0.2.0

### PR DESCRIPTION
**Pull request by fblundun**

### CODE REVIEW

File: 2-repositories/scala-repo-server/src/test/scala/com.snowplowanalytics.iglu.server/service/SchemaServiceSpec.scala [hunk 1]
```diff
@@ -226,93 +190,93 @@ class SchemaServiceSpec extends Specification
...
+
+        // "return a proper json for multi vendor urls with public schemas" +
+        // s"(${multiVendorPublic})" in {
+        //   Get(multiVendorPublic) ~> addHeader("apikey", readKey) ~> routes ~>
+        //   check {
+        //     status === OK
+        //     responseAs[String] must contain(otherVendor) and
+        //       contain(otherVendor2)
+        //   }
+        // }
```
Comment (alexanderdean) [inline 2-repositories/scala-repo-server/src/test/scala/com.snowplowanalytics.iglu.server/service/SchemaServiceSpec.scala]: Any reason why these are commented out not deleted?

File: 2-repositories/scala-repo-server/src/test/scala/com.snowplowanalytics.iglu.server/service/SchemaServiceSpec.scala [hunk 2]
```diff
@@ -328,119 +292,119 @@ class SchemaServiceSpec extends Specification
...
+
+        // "return proper metadata for multi vendor urls with public schemas" +
+        // s"(${metaMultiVendorPublic})" in {
+        //   Get(metaMultiVendorPublic) ~> addHeader("apikey", readKey) ~>
+        //   routes ~> check {
+        //     status === OK
+        //     responseAs[String] must contain(otherVendor) and
+        //       contain(otherVendor2)
+        //   }
+        // }
```
Comment (alexanderdean) [inline 2-repositories/scala-repo-server/src/test/scala/com.snowplowanalytics.iglu.server/service/SchemaServiceSpec.scala]: Why commented out?

File: 2-repositories/scala-repo-server/src/test/scala/com.snowplowanalytics.iglu.server/service/SchemaServiceSpec.scala [hunk 3]
```diff
@@ -796,40 +760,40 @@ class SchemaServiceSpec extends Specification
...
+      //     }
+      // }
+
+      // "return a 401 if the schema already exists with body request" in {
+      //   Post(postUrl12, HttpEntity(`application/json`, validSchema)) ~>
+      //   addHeader("apikey", writeKey) ~> sealRoute(routes) ~> check {
+      //     status === Unauthorized
+      //     responseAs[String] must contain("This schema already exists")
+      //   }
+      // }
```
Comment (alexanderdean) [inline 2-repositories/scala-repo-server/src/test/scala/com.snowplowanalytics.iglu.server/service/SchemaServiceSpec.scala]: Why commented out?

File: 2-repositories/scala-repo-server/src/test/scala/com.snowplowanalytics.iglu.server/service/SchemaServiceSpec.scala [hunk 4]
```diff
@@ -1012,36 +976,36 @@ class SchemaServiceSpec extends Specification
...
+
+      // "return a 200 if the schema already exists with request body" in {
+      //   Put(postUrl11, HttpEntity(`application/json`, validSchema)) ~>
+      //   addHeader("apikey", writeKey) ~> sealRoute(routes) ~> check {
+      //     status === OK
+      //     responseAs[String] must contain("Schema successfully updated") and
+      //       contain(vendor)
+      //   }
+      // }
```
Comment (alexanderdean) [inline 2-repositories/scala-repo-server/src/test/scala/com.snowplowanalytics.iglu.server/service/SchemaServiceSpec.scala]: Why commented out?

**Comment by alexanderdean** [comment]
It's looking good Fred! Can you reword the commits please to match up with the ticket names? Currently the commits look like this:

```
Prepared for release - WIP
Common: add .travis.yml file to repo (close #113)
Documentation: add Travis, release and license buttons to README (close #104)
Repo Syncer: add (close #77)
Documentation: add Objective-C client to 1-clients/README.md (close #101)
Objective-C Client: add git submodule (close #102)
Bumped version to 0.2.0
Scala Repo Server: make app crash instead of hanging when run with no arguments (close #106)
Common: add Vagrant push script to publish Scala Repo Server (close #112)
Common: add dedicated Vagrant setup (close #96)
Added ability to load configuration from environment variables (closes #95)
Changed header name from api_key to apikey so it survives ELB (closes #94)
Ensured that createdat and updatedat are the same when a schema is first uploaded (fixes #92)
Hotfix making it possible to read public schemas without an API key (closes #100)
Started returning 200 when multiple schemas are deleted (closes #99)
Stopped throwing exception when attempting to delete a nonexistent schema (fixes #90)
Scala Repo Server: make it possible to create * auth keys (close #72)
Scala Repo Server: fix setup process for test suite (fix #79)
Removed .test from test class packages (closes #89)
Fixed schema vendor/name/format/version regexen (fixes #81)
Sublime Text cannot comprehend quadruple quotes
Updated test suite to work with new bootstrap logic
Ensured that the self-describing schema gets created on startup (closes #65)
Made attributes of self-describing schema consistent with Iglu Central (closes #98)
creates the schema needed for validating self-describing schemas when launching the server for the first time #65
Added ability to delete schemas (closes #87)
Added startup and teardown scripts to run before and after tests
Updated json4s dependency to 3.2.11 (closes #78)
```

File: 2-repositories/scala-repo-server/src/main/scala/com.snowplowanalytics.iglu.server/util/ServerConfig.scala [hunk 5]
```diff
@@ -28,16 +31,18 @@ object ServerConfig {
...
   //Port on which the server will be running
-  val port = config.getInt("repo-server.port")
+  val port = env("PORT").map(_.toInt) getOrElse config.getInt("repo-server.port")
 
-  val pgHost = config.getString("postgres.host")
-  val pgPort = config.getInt("postgres.port")
-  val pgDbName = config.getString("postgres.dbname")
-  val pgUsername = config.getString("postgres.username")
-  val pgPassword = config.getString("postgres.password")
+  val pgHost = env("RDS_HOSTNAME") getOrElse config.getString("postgres.host")
```
Comment (alexanderdean) [inline 2-repositories/scala-repo-server/src/main/scala/com.snowplowanalytics.iglu.server/util/ServerConfig.scala]: These environment variable names tie us to Amazon RDS unnecessarily. A user could want to provide their Postgres connection details via environment variables regardless of whether they are using Amazon RDS or not...

Comment (fblundun) [inline 2-repositories/scala-repo-server/src/main/scala/com.snowplowanalytics.iglu.server/util/ServerConfig.scala]: Good point - how about changing `RDS_HOSTNAME -> DB_HOSTNAME` and so on?

Comment (alexanderdean) [inline 2-repositories/scala-repo-server/src/main/scala/com.snowplowanalytics.iglu.server/util/ServerConfig.scala]: I think `PG_` would be a bit more futureproof (for adding `ZK_HOSTNAME` or similar in the future).

Also, is it bad form for us not to be "vendoring" our environment variables - e.g. `IGLU_`?

Comment (fblundun) [inline 2-repositories/scala-repo-server/src/main/scala/com.snowplowanalytics.iglu.server/util/ServerConfig.scala]: I wouldn't be surprised if somebody else is already using `PG_HOSTNAME` as an environment variable name.

So let's go with `IGLU_PG_HOSTNAME`.

**Comment by alexanderdean** [comment]
Thanks @fblundun - if you could do the new env var names and the commit rebasing, then I will do another build for testing!

**Comment by fblundun** [comment]
Ok! I'll let you know when it's ready.

**Comment by fblundun** [comment]
@alexanderdean ready now!

**Comment by alexanderdean** [comment]
Thanks @fblundun ! Looks good...

File: CHANGELOG [hunk 6]
```diff
@@ -1,3 +1,35 @@
+Release 2 Penny Black (2014-xx-xx)
```
Comment (alexanderdean) [inline CHANGELOG]: Typo, should be "Release 3"

**Pull request merged**